crates/aptos-crypto/src/p256_ecdsa/mod.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+//! This module provides an API for the ECDSA signature scheme over the NIST-P256 curve as defined in [NIST SP 800-186](https://csrc.nist.gov/publications/detail/sp/800-186/final).
+//!
+//! Signature verification also checks and rejects non-canonical signatures. Signing is guaranteed
+//! to output the canonical signature which passes this module's verification.
+//!
+//! # Examples
+//!
+//! ```
+//! use aptos_crypto_derive::{CryptoHasher, BCSCryptoHash};
+//! use aptos_crypto::{
+//!     p256_ecdsa::*,
+//!     traits::{Signature, SigningKey, Uniform},
+//!     test_utils::KeyPair
+//! };
+//! use rand::{rngs::StdRng, SeedableRng};
+//! use rand_core::OsRng;
+//! use serde::{Serialize, Deserialize};
+//!
+//! #[derive(Serialize, Deserialize, CryptoHasher, BCSCryptoHash)]
+//! pub struct TestCryptoDocTest(String);
+//! let message = TestCryptoDocTest("Test message".to_string());
+//!
+//! let mut rng = OsRng;
+//! let kp = KeyPair::<P256PrivateKey, P256PublicKey>::generate(&mut rng);
+//!
+//! let signature = kp.private_key.sign(&message).unwrap();
+//! assert!(signature.verify(&message, &kp.public_key).is_ok());
+//! ```
+
+/// The length in bytes of the P256PrivateKey
+pub const P256_PRIVATE_KEY_LENGTH: usize = 32;
+/// The length in bytes of the P256PublicKey
+pub const P256_PUBLIC_KEY_LENGTH: usize = 65;
+/// The length in bytes of the P256Signature
+pub const P256_SIGNATURE_LENGTH: usize = 64;
+
+/// The order of P256 as defined in [NIST SP 800-186](https://csrc.nist.gov/publications/detail/sp/800-186/final)
+const ORDER: [u8; 32] = [
```

Comment:
Clarify if this is in LE or BE encoding...

Reply:
Big-endian. Comment added